src/two_sum.rs

```diff
@@ -0,0 +1,61 @@
+use std::collections::HashMap;
+/*
+ *  From the leetcode: https://leetcode.com/problems/two-sum/description/
+ *  I did not get to choose the signature here.  I believe it should return (usize, usize)...
+ *  or even Option<(usize, usize)>.
+ */
+pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
+    /*
+     *  Per the Constraints of the problem, there will only ever be exactly one solution.
+     *  incidentally, this means that either there will only be exactly _two_ of a number, or
+     *  that the number, doubled, does not equal the target.  In either case, I only then need
+     *  a single index for a given value, as during sum finding, it either won't matter, or
+     *  whichever one I _don't_ store will just be found pointing to the one I _did store.
+     */
+    let lookup: HashMap<i32, usize> = nums.iter().copied().enumerate().map(swap).collect();
+
+    let (idx, first) = nums
+        .into_iter()
+        .enumerate()
+        .find(|(i, x)| lookup.get(&(target - x)).map_or(false, |j| j != i))
+        /*
+         *  Blech.  But the constraint is there _will_ be a solution, so the return
+         *  type does not allow for Monads
+         */
+        .unwrap();
+
+    [idx, lookup[&(target - first)]]
+        .into_iter()
+        .map(|i| i as i32) // Don't love this.
```

Comment:
Yeah, I hear you. If you want to skip the blind cast, the code will _almost certainly_ be longer, one way or the other, because a `usize` is bigger than an `i32` so Rust _wants_ you to "do it the right way" and use a conversion method that can fail. 

Broadly, this means switching from a cast to a [conversion that can fail](https://doc.rust-lang.org/std/primitive.i32.html#impl-TryFrom%3Cusize%3E-for-i32):

```rust
[idx, lookup[&(target - first)]
  .into_iter()
  .map(i32::try_from)
```

From here, it becomes more a question of how you want to do the type juggling to handle your `Iterator<Item::Result<i32, error>>`. One absolutely _wild_ option: you can `collect` a `Vec<Result>`... into a `Result<Vec>` :exploding_head: 

```rust
let final: Result<Vec<i32>, _> = [idx, lookup[&(target - first)]
  .into_iter()
  .map(i32::try_from)
  .collect();
```

...which you can then just return, or match/unpack/`expect`/whatever. 

In contexts like this, and Advent of Code, when I have some reason to believe things Should Work -- I trust the inputs, I know a lot about the data, that kind of thing, I often use `filter_map`:

```rust
[idx, lookup[&(target - first)]
  .into_iter()
  .filter_map(|num| i32::try_from(num).ok()) // Convert our Result to an Option using `.ok`, then filter for only the `Some`s.
  .collect()
```

...it's also 100% valid to just cast. Entirely up  to you, this one is all style.

Reply:
So... I beat the blind cast by fixing the signature.  If you're looking for indices, usize seems reasonable.